nomad-types/src/macros.rs

```diff
@@ -0,0 +1,53 @@
+#[macro_export]
+/// Implement deser_nomad_number for a uint type
+macro_rules! impl_deser_nomad_number {
+    ($($u:ident),*) => {
+        $(paste::paste! {
+            #[doc = "Permissive deserialization of numbers. Allows numbers, hex strings, and decimal strings"]
+            pub fn [<deser_nomad_number_ $u>]<'de, D>(deserializer: D) -> Result<$u, D::Error>
+            where
+                D: serde::Deserializer<'de>,
+            {
+                struct NumberOrNumberStringVisitor;
+
+                impl<'de> serde::de::Visitor<'de> for NumberOrNumberStringVisitor {
+                    type Value = $u;
+
+                    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                        formatter
+                            .write_str("an integer, a decimal string, or a 0x-prepended hexadecimal string")
+                    }
+
+                    fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
```

Comment:
I believe all numbers in serde::Deserialize are `u64` by default. I previously had visit_u32, visit_u16, etc... but tests were failing and deserializing to any type other than `u64` was panicking.

Reply:
Ahh I see, all the non-u64 visit methods forward to `visit_u64`: https://docs.serde.rs/serde/de/trait.Visitor.html#method.visit_u32